internal-packages/next-dev/src/durableObjects.ts

```diff
@@ -41,31 +46,33 @@ export async function getDOBindingInfo(
 	}
 
 	if (!registeredWorkers) {
-		warnAboutLocalDurableObjectsNotFound(requestedDurableObjectsNames);
+		warnAboutLocalDurableObjectsNotFound(requestedDurableObjectBindingNames);
 		return;
 	}
 
 	const registeredWorkersWithDOs: RegisteredWorkersWithDOs =
-		getRegisteredWorkersWithDOs(
-			registeredWorkers,
-			requestedDurableObjectsNames,
-		);
+		getRegisteredWorkersWithDOs(registeredWorkers);
 
 	const [foundDurableObjects, missingDurableObjects] = [
-		...requestedDurableObjectsNames.keys(),
+		...requestedDurableObjectBindingNames.keys(),
 	].reduce(
-		([foundDOs, missingDOs], durableObjectName) => {
+		([foundDOs, missingDOs], durableObjectBindingName) => {
 			let found = false;
-			for (const [, worker] of registeredWorkersWithDOs.entries()) {
-				found = !!worker.durableObjects.find(
-					durableObject => durableObject.name === durableObjectName,
-				);
+			for (const [workerName, worker] of registeredWorkersWithDOs.entries()) {
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				const durableObject = durableObjects[durableObjectBindingName]!;
+				if (workerName === durableObject.service.name) {
+					found = !!worker.durableObjects.find(
+						durableObject =>
+							durableObject.className === durableObject.className,
```

Comment:
> This will always be true!

Yes, such a silly mistake! 😓 
(basically I reused the same variable name and didn't notice it 🤦)

___

> I think this whole block of code could be simpler.
Thanks, I've applied the suggested change, I agree that it looks better 😄 
(I was going a bit overboard with the `reduce` there 🙈)